src/main/java/org/jabref/gui/linkedfile/LinkedFileDialogController.java

```diff
@@ -27,26 +30,46 @@ public class LinkedFileEditDialogView extends BaseDialog<LinkedFile> {
 
     @Inject private DialogService dialogService;
     @Inject private StateManager stateManager;
-
     @Inject private PreferencesService preferences;
 
     private LinkedFilesEditDialogViewModel viewModel;
-
     private final LinkedFile linkedFile;
 
-    public LinkedFileEditDialogView(LinkedFile linkedFile) {
+    /**
+     * Constructor for adding a new LinkedFile.
+     */
+    public LinkedFileDialogController() {
+        this(new LinkedFile("", "", ""));
+        initializeDialog(Localization.lang("Add file link"), ADD_BUTTON);
+    }
+
+    /**
+     * Constructor for editing an existing LinkedFile.
+     *
+     * @param linkedFile The linked file to be edited.
+     */
+    public LinkedFileDialogController(LinkedFile linkedFile) {
         this.linkedFile = linkedFile;
+        initializeDialog(Localization.lang("Edit file link"), EDIT_BUTTON);
+    }
```

Comment:
Logic for editing file doesn't seem to be present. If this is a unified Linked File Dialog controller, the logic for both adding and editing should be present.

Reply:
@subhramit I have fixed the localization key error and will commit the changes.

Regarding the logic for adding and editing, the LinkedFileDialogController already contains unified logic for both adding and editing LinkedFile objects. The constructors differentiate between the two actions:

Adding a new file: The no-argument constructor initializes a new LinkedFile and sets up the dialog with the "Add file link" title and the Add button.
Editing an existing file: The constructor that accepts a LinkedFile object sets up the dialog with the "Edit file link" title and the Apply button.
Both actions use the same initializeDialog and initialize methods, which handle the setup and binding of UI components to the viewModel. The result converter ensures that the correct LinkedFile is returned when the dialog is closed. Thus, the unified logic for both adding and editing is present and functioning correctly.

I have verified the functionality by adding, editing, saving, and reopening the file. I am able to both add and edit the file without any issues.

Please let me know if there is something I am missing.

Reply:
Ah okay you've extracted a function and called it. The confusion happens because of names. You've renamed the view into a controller, will need some thought on whether this needs conformation to relevant existing architecture. I'll leave another review and let the lead maintainers take over.